mailboxresource.py

```diff
@@ -34,13 +32,19 @@ def copy_emails(self, days, local_folder, wkhtmltopdf):
             criterion = '(SENTSINCE {date})'.format(date=date)
 
         typ, data = self.mailbox.search(None, criterion)
+
+        if has_progress:
+            from progress.bar import ShadyBar
+            progress_bar = ShadyBar('Downloading messages', max=len(data[0].split()), suffix='%(percent).1f%% | %(remaining).0f Emails left | Estimated %(eta)d seconds remaining')
```

Comment:
Please, can you put the `data[0].split()` result into a variable before the `if has_progress` and use the variable for progress bar and the `for` loop. I think this can be an expensive operation.